common/lib/elf.c

```diff
@@ -37,28 +37,33 @@
 
 #define DF_1_PIE    0x08000000
 
-#define ABI_SYSV     0x00
-#define ARCH_X86_64  0x3e
-#define ARCH_X86_32  0x03
-#define ARCH_AARCH64 0xb7
-#define ARCH_RISCV   0xf3
-#define BITS_LE      0x01
-#define ELFCLASS64   0x02
-#define SHT_RELA     0x00000004
-#define R_X86_64_NONE      0x00000000
-#define R_AARCH64_NONE     0x00000000
-#define R_RISCV_NONE       0x00000000
-#define R_X86_64_RELATIVE  0x00000008
-#define R_AARCH64_RELATIVE 0x00000403
-#define R_RISCV_RELATIVE   0x00000003
-#define R_X86_64_GLOB_DAT  0x00000006
-#define R_AARCH64_GLOB_DAT 0x00000401
-#define R_X86_64_JUMP_SLOT 0x00000007
+#define ABI_SYSV            0x00
+#define ARCH_X86_64         0x3e
+#define ARCH_X86_32         0x03
+#define ARCH_AARCH64        0xb7
+#define ARCH_RISCV          0xf3
+#define ARCH_LOONGARCH      0x102
+#define BITS_LE             0x01
+#define ELFCLASS64          0x02
+#define SHT_RELA            0x00000004
+#define R_X86_64_NONE       0x00000000
+#define R_AARCH64_NONE      0x00000000
+#define R_RISCV_NONE        0x00000000
+#define R_LARCH_NONE        0x00000000
+#define R_X86_64_RELATIVE   0x00000008
+#define R_AARCH64_RELATIVE  0x00000403
+#define R_RISCV_RELATIVE    0x00000003
+#define R_LARCH_RELATIVE    0x00000003
+#define R_X86_64_GLOB_DAT   0x00000006
+#define R_AARCH64_GLOB_DAT  0x00000401
+#define R_X86_64_JUMP_SLOT  0x00000007
 #define R_AARCH64_JUMP_SLOT 0x00000402
-#define R_RISCV_JUMP_SLOT  0x00000005
-#define R_X86_64_64        0x00000001
-#define R_RISCV_64         0x00000002
-#define R_AARCH64_ABS64    0x00000101
+#define R_RISCV_JUMP_SLOT   0x00000005
+#define R_LARCH_JUMP_SLOT   0x00000005
+#define R_X86_64_64         0x00000001
+#define R_RISCV_64          0x00000002
+#define R_LARCH_64          0x00000002
+#define R_AARCH64_ABS64     0x00000101
 
```

Comment:
Please don't change the alignment of other variables, even if they don't align with the ones you are adding, in order to not clutter the diff.